### PR TITLE
feat(EMS-548): Create and get an application

### PR DESCRIPTION
**Pull request by ttbarnes**
This PR creates an application after completing insurance eligibility and adds the ability to display the application in the main Application page, "All sections". The next PR will replace the example data on this page with a task-list.

## Summary of changes

- Docker build from another PR: https://github.com/UK-Export-Finance/exip/pull/165
- Add `SESSION_SECRET` to docker build environment variables/args

- When completing the insurance eligibility (meaning, pressing "continue" on the "Eligible to apply online" page), create an application.
- Add "all sections" page and controller. When visiting the page, fetch the application from API/DB.

- Split up the UI api file to handle the following interactions from the API/DB:
  - Creating an empty application.
  - Getting and updating an application.
  - Updating eligibility
  - Getting a country in the database by ISO code.
  - Improve error handling.

- Add the first set of API/keystone tests (`src/api/.keystone/keystone.test.ts`)

## Data 
- When creating an application:
  - Add default data such as reference number, submission deadline, created/updated date
  - Create and update an eligibility entry in the database with the provided answers.
- When an application is created:
  - Add the application ID to the reference number entry.
- All of the above happens in `src/api/schema.ts`.

## Other changes
- Rename `haveCompaniesHouseNumber` to `hasCompaniesHouseNumber`.
- Add `generated_reports` to `.gitignore`
- Linting and rebuild UI bundled scripts/styles due to `package-lock` changes.

### CODE REVIEW

File: src/api/babel.config.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module.exports = {
+  presets: [
+    [
+      '@babel/preset-env',
+      {
+        targets: {
+          node: 10,
+          browsers: ['last 2 chrome versions', 'last 2 firefox versions', 'last 2 safari versions', 'last 2 edge versions'],
```
Comment (Zainzzkk) [inline src/api/babel.config.js:8]: not important but quick question, what happens for users on older browsers?

File: src/ui/server/constants/routes/insurance.ts [hunk 2]
```diff
@@ -1,26 +1,27 @@
 import { PRODUCT } from '../product';
 
-const INSURANCE_ROOT = '/insurance';
+export const ROOT = '/insurance';
```
Comment (Zainzzkk) [inline src/ui/server/constants/routes/insurance.ts:3]: Is not having this as INSURANCE_ROOT better? If we export this anywhere (and have root for quote as well)

File: src/ui/server/controllers/insurance/all-sections/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+      const expectedVariables = {
+        ...insuranceCorePageVariables({
+          PAGE_CONTENT_STRINGS: PAGES.INSURANCE.START,
+          BACK_LINK: req.headers.referer,
+        }),
+        application: mockApplication,
+      };
+
+      expect(res.render).toHaveBeenCalledWith(TEMPLATES.INSURANCE.ALL_SECTIONS, expectedVariables);
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/all-sections/index.test.ts:52]: minor but could we dereference templates on the top and make them super neat, then we can also only have to change in one place if the template name changes?

File: src/ui/server/controllers/insurance/eligibility/eligible-to-apply-online/index.test.ts [hunk 4]
```diff
@@ -30,10 +38,52 @@ describe('controllers/insurance/eligibility/eligible-to-apply-online', () => {
   });
 
   describe('post', () => {
-    it(`should redirect to ${ROUTES.INSURANCE.ELIGIBILITY.ALREADY_HAVE_ACCOUNT}`, () => {
```
Comment (Zainzzkk) [inline src/ui/server/controllers/insurance/eligibility/eligible-to-apply-online/index.test.ts:33]: can we dereference these?

Comment (ttbarnes) [reply src/ui/server/controllers/insurance/eligibility/eligible-to-apply-online/index.test.ts:33]: This has been removed? It's now `${ROOT}/${mockReferenceNumber}${INSURANCE_ROUTES.ALL_SECTIONS}`

File: src/ui/server/constants/routes/insurance.ts (hunk 2 above)
Comment (ttbarnes) [reply src/ui/server/constants/routes/insurance.ts:3]: Good shout, updated.

File: src/ui/server/controllers/insurance/eligibility/eligible-to-apply-online/index.test.ts (hunk 4 above)
Comment (Zainzzkk) [reply src/ui/server/controllers/insurance/eligibility/eligible-to-apply-online/index.test.ts:33]: ignore me, my bad

File: src/ui/server/controllers/insurance/all-sections/index.test.ts (hunk 3 above)
Comment (ttbarnes) [reply src/ui/server/controllers/insurance/all-sections/index.test.ts:52]: Great idea, have added `TEMPLATE` to this controller and we can use this approach for all controllers moving forward? Will create a ticket to update previously created controllers to use this approach

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Review by Zainzzkk — APPROVED** [review]
LGTM

**Review by Zainzzkk — APPROVED** [review]
LGTM

File: .github/workflows/pr_tests.yml [hunk 5]
```diff
@@ -84,21 +84,35 @@ jobs:
...
+    # Cancel Previous runs
+    # concurrency:
+    #   group: e2e-${{ github.workflow }}-${{ github.ref }}-${{ matrix.containers }}
+    #   cancel-in-progress: true
 
     steps:
       - uses: actions/checkout@v3
       - name: Start Docker Compose
+        env:
+          API_URL: ${{ secrets.API_URL }}
```
Comment (IgnatG) [inline .github/workflows/pr_tests.yml:96]: wasn't this added/moved in a different PR?

File: e2e-tests/cypress/e2e/journeys/insurance/eligibility/eligible-to-apply-online.spec.js [hunk 6]
```diff
@@ -109,12 +109,16 @@ context('Insurance - Eligibility - You are eligible to apply online page - I wan
   });
 
   describe('form submission', () => {
-    it(`should redirect to ${ROUTES.INSURANCE.ELIGIBILITY.ALREADY_HAVE_ACCOUNT}`, () => {
+    it(`should redirect to ${ROUTES.INSURANCE.ROOT}/[referenceNumber]${ROUTES.INSURANCE.ALL_SECTIONS}`, () => {
```
Comment (IgnatG) [inline e2e-tests/cypress/e2e/journeys/insurance/eligibility/eligible-to-apply-online.spec.js:112]: same as above - technically, if it was already in main, then this should be here?

File: .github/workflows/pr_tests.yml (hunk 5 above)
Comment (ttbarnes) [reply .github/workflows/pr_tests.yml:96]: Yes, that PR was merged into this PR instead of `main-application` unfortunately

File: src/ui/server/graphql/apollo.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ dotenv.config();
 const urlRoot = process.env.API_URL;
 
 const createHttpLinkOptions = {
-  uri: `${urlRoot}/graphql`,
+  uri: `${urlRoot}`,
```
Comment (IgnatG) [inline src/ui/server/graphql/apollo.ts:13]: same as the others - this was in a different PR?

File: e2e-tests/cypress/e2e/journeys/insurance/eligibility/eligible-to-apply-online.spec.js (hunk 6 above)
Comment (ttbarnes) [reply e2e-tests/cypress/e2e/journeys/insurance/eligibility/eligible-to-apply-online.spec.js:112]: The other PR was merged into this PR unfortunately. The build issues stemed from this PR as it's the first API integration branch

**Review by IgnatG — APPROVED** [review]
Great work!
